app/code/Magento/CompareListGraphQl/Model/Resolver/AddProductsToCompareList.php

```diff
@@ -0,0 +1,95 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\CompareListGraphQl\Model\Resolver;
+
+use Magento\Catalog\Model\MaskedListIdToCompareListId;
+use Magento\CompareListGraphQl\Model\Service\AddToCompareList;
+use Magento\CompareListGraphQl\Model\Service\GetCompareList;
+use Magento\Framework\GraphQl\Config\Element\Field;
+use Magento\Framework\GraphQl\Exception\GraphQlInputException;
+use Magento\Framework\GraphQl\Query\Resolver\ContextInterface;
+use Magento\Framework\GraphQl\Query\Resolver\Value;
+use Magento\Framework\GraphQl\Query\ResolverInterface;
+use Magento\Framework\GraphQl\Schema\Type\ResolveInfo;
+
+/**
+ * Add products item to compare list
+ */
+class AddProductsToCompareList implements ResolverInterface
+{
+    /**
+     * @var AddToCompareList
+     */
+    private $addProductToCompareList;
+
+    /**
+     * @var GetCompareList
+     */
+    private $getCompareList;
+
+    /**
+     * @var MaskedListIdToCompareListId
+     */
+    private $maskedListIdToCompareListId;
+
+    /**
+     * @param AddToCompareList $addProductToCompareList
+     * @param GetCompareList $getCompareList
+     * @param MaskedListIdToCompareListId $maskedListIdToCompareListId
+     */
+    public function __construct(
+        AddToCompareList $addProductToCompareList,
+        GetCompareList $getCompareList,
+        MaskedListIdToCompareListId $maskedListIdToCompareListId
+    ) {
+        $this->addProductToCompareList = $addProductToCompareList;
+        $this->getCompareList = $getCompareList;
+        $this->maskedListIdToCompareListId = $maskedListIdToCompareListId;
+    }
+
+    /**
+     * Add products to compare list
+     *
+     * @param Field $field
+     * @param ContextInterface $context
+     * @param ResolveInfo $info
+     * @param array|null $value
+     * @param array|null $args
+     *
+     * @return Value|mixed|void
+     *
+     * @throws GraphQlInputException
+     * @SuppressWarnings(PHPMD.UnusedFormalParameter)
+     */
+    public function resolve(
+        Field $field,
+        $context,
+        ResolveInfo $info,
+        array $value = null,
+        array $args = null
+    ) {
+        $storeId = (int)$context->getExtensionAttributes()->getStore()->getStoreId();
+        if (!isset($args['input']['uid'])) {
```

Comment:
This check will be performed automatically on the schema level. We should rather check if the value is not empty. Please, check other similar places as well

Reply:
**Done**